src/binary_tree/interfaces.ts

```diff
@@ -0,0 +1,40 @@
+export interface IBinaryNode<T> {
```

Comment:
So, previous we put interfaces near the realisation, now all interfaces in the separate file. What's the point of this changes?